lib/common.js

```diff
@@ -343,13 +345,132 @@ var readTemporalFraction = function(parser, fractionPrecision) {
   }
 
   return {
-    value: fraction,
-    precision: fractionPrecision,
-    milliseconds: milliseconds
+    value: fraction,              // the integer after the decimal place
+    precision: fractionPrecision, // the number of digits after the decimal
+    milliseconds: milliseconds    // the unrounded 3 digits after the decimal
   };
 };
 
-exports.readMysqlValue = function(parser, column, columnSchema, tableMap, emitter) {
+// fraction is the fractional second object from readTemporalFraction().
```

Comment:
This is a lot of code for this file now, is it possible to move the date/time specific functions to something like `lib/datetime_decode.js` similar to the `json_decode.js`? Also, there are date/time/timestamp opaque values in the JSON type that probably need modification too.

Reply:
Done.

Reply:
Unfortunately I can't easily work on the JSON code, as my version of MariaDB doesn't support JSON.  Now that it is in a different file, it should be easy to reuse code if you want.  It appears that you are already outputting dateStrings in the JSON code whether or not the dateStrings option is in use because JSON doesn't include Date objects, so I am not sure that you need to do that.